File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        ans=[]
        d=0
        for i in s:
            if i=='(':
                if d>0:
                    ans.append(i)
                d+=1
            else:
                d-=1
                if d>0:
                    ans.append(i)

        return ''.join(ans)





           
            

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
    int f=0;
    Stack<Character>st=new Stack<>();
    char[] ch=s.toCharArray();

    for(char c:ch){
        if(c=='('){
            if(f>0){
                st.push(c);
            }
            f++;          
        }
        else{
            f--;
            if(f>0){
                st.push(c);
            }
        }
    }
    
    StringBuilder sb=new StringBuilder();
    for(char c:st)sb.append(c);
    return sb.toString();

    
    
    }
}